java/ql/lib/semmle/code/java/Expr.qll

```diff
@@ -296,19 +296,20 @@ class CompileTimeConstantExpr extends Expr {
    *
    * Note that this does not handle the following cases:
    *
-   * - values of type `long`,
-   * - `char` literals.
+   * - values of type `long`.
    */
   cached
   int getIntValue() {
     exists(IntegralType t | this.getType() = t | t.getName().toLowerCase() != "long") and
     (
       exists(string lit | lit = this.(Literal).getValue() |
-        // `char` literals may get parsed incorrectly, so disallow.
+        // Don't parse `char` literal as int, instead get its code point value (see below)
         not this instanceof CharacterLiteral and
         result = lit.toInt()
       )
       or
+      result = this.(CharacterLiteral).getCodePointValue()
```

Comment:
We probably don't want to surprise users by exposing this change through an existing API

Reply:
I am not sure if this is really a surprise. If you perform an arithmetic operation with a char literal in Java then it actually uses its code point, e.g. `'a' + 1` is 98.
If a new predicate was added for this, then a lot of the logic for the arithmetic operations of CompileTimeConstantExpr would have to be duplicated.

Additionally `getIntValue()` already covers a cast to `char`, so there is not a big difference then.